jenkins_init.ps1

```diff
@@ -0,0 +1,78 @@
+# Assign the first argument passed to the script to the variable TUTORIAL
+$TUTORIAL = $args[0]
+
+# Assign values 1-4 to variables VAR1-VAR4 can be changed in the future to Maven, Node.js, Python, etc.
+$VAR1 = 1
+$VAR2 = 2
+$VAR3 = 3
+$VAR4 = 4
+
+# Assign file paths to variables VAR0-VAR3L
+$VARDL = "."
+$VAR1L = "./experimental_docker_compose_files/00_old_one_from_proposal"
+$VAR2L = "./experimental_docker_compose_files/01_simple_controller_plus_agent"
+$VAR3L = "./experimental_docker_compose_files/02_custom_docker_file_connecting_agent_and_controller"
+$VAR4L = "./experimental_docker_compose_files/03_maven_tutorial"
+$DOCKER_COMPOSE = "docker-compose"
+
+# Function to check if tutorials are already running
+function Check-RunningTutorials {
+    # Check if there is a .tutorials_running.txt file, if there is then check if it's empty, if not stops the script
+    if (-not (Test-Path -Path ".\.tutorials_running.txt")) {
+        Write-Host "Running First Tutorial"
+    }
+    else {
+        if ((Get-Content -Path ".\.tutorials_running.txt" | Measure-Object).Count -gt 0) {
+            Write-Host "Another Tutorial is running. Please use ./jenkins_teardown.ps1 first"
+            exit 1
+        }
+    }
+}
+
+# Function to generate SSH keys
+function Generate-SSHKeys {
+    param(
+        [string]$tutorialPath
+    )
+
+    Write-Host "generating new ssh keys"
+    & $tutorialPath\keygen.ps1 $tutorialPath
+}
+
+# Function to start a tutorial based on the provided path
+function Start-Tutorial {
+    param(
+        [string]$tutorialPath
+    )
+
+    Write-Host "Starting tutorial $tutorialPath"
+    & $DOCKER_COMPOSE -f "$tutorialPath/docker-compose.yaml" up -d
+}
+
+# Check if tutorials are already running
+Check-RunningTutorials
```

Comment:
Compared with the bash one.

https://github.com/ash-sxn/GSoC-2023-docker-based-quickstart/blob/main/jenkins_init.sh#L62

I think we also need to check if docker and docker-compose exist.